classes/cron_processor.php

```diff
@@ -142,12 +144,18 @@ public function findtaskname(\ExcimerLogEntry $sample): ?string {
      * @throws \dml_exception
      */
     public function process(manager $manager, float $finishtime): void {
+        $reasonstack = 0;
         $duration = $finishtime - $this->currenttask->starttime;
+
+        if (property_exists($this, 'currenttask') && $this->currenttask
```

Comment:
can you move this logic into get_reasons() it should be fairly similar to 

$this->is_considered_slow($profile)

ala:

$this->is_considered_deep($profile)